instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -731,7 +731,9 @@ def _instrument(self, **kwargs):
         self._original_async_client = httpx.AsyncClient
         request_hook = kwargs.get("request_hook")
         response_hook = kwargs.get("response_hook")
-        async_request_hook = kwargs.get("async_request_hook", request_hook)
```

Comment:
I'm wondering if we should remove this logic of defaulting to `request hook` if `async_request_hook` does not exist. It doesn't make much intuitive sense to me. We can then probably just add a check to see if the hook coming from `async_request_hook` is indeed an async function with `inspect.iscoroutinefunction` as part of the [inspect](https://docs.python.org/3/library/inspect.html) module and not assign it to `_InstrumentedAsyncClient._request_hook` if not.

Reply:
The easiest way is probably to leave it as None; it will fail during the callable check when hook is None. Also, this PR needs the async_response_hook fix as well. And of course, some tests would be nice to avoid problems with this again. Wdyt?

Reply:
> leave it as None if not set; it will fail during the callable check when the hook is None.

Yes this and checking if `async_request_hook` is actually `async` is probably sufficient, and as always, adding more tests are always welcome.

Reply:
Thanks for your review. I will work on the changes.